yarn-project/noir-libs/noir-aztec/src/state_vars/set.nr

```diff
@@ -22,16 +23,21 @@ impl<Note, N> Set<Note, N> {
         create_note(context, self.storage_slot, note, self.note_interface);
     }
 
-    fn insert_from_public(self, note: Note) {
-        let compute_note_hash = self.note_interface.compute_note_hash;
-        let note_hash = compute_note_hash(note);
-        create_commitment(note_hash);
+    fn insert_from_public(self, inputs: PublicContextInputs, note: &mut Note) {
+        create_note_hash_from_public(inputs, self.storage_slot, note, self.note_interface);
     }
 
     fn assert_contains(self, context: &mut Context, note: &mut Note) {
         ensure_note_exists(context, self.storage_slot, self.note_interface, note);
     }
 
+    // TODO(https://github.com/AztecProtocol/aztec-packages/issues/1386): rename to
+    // `assert_contains` and replace function above ^ once public kernel injects
+    // nonces to note hashes.
+    fn assert_contains_note_hash(self, context: &mut Context, note: &mut Note) {
```

Comment:
assert_note_hash is maybe enough?

Reply:
:thinking: maybe but I'm mimicking assert the above function `assert_contains`. Eventually we can replace `assert_contains` with this implementation. Right now `assert_contains_note_hash` only works for notes created in public.